akka-stream/src/main/scala/akka/stream/Queue.scala

```diff
@@ -35,3 +43,29 @@ trait SinkQueue[T] {
    */
   def pull(): Future[Option[T]]
 }
+
+sealed trait StreamCallbackStatus[T]
```

Comment:
I'm going to add this to Sink.queue. It's currently Future[Option[Message]] which is not descriptive enough I beleive